sound/soc/soc-pcm.c

```diff
@@ -2502,18 +2502,13 @@ static int dpcm_fe_dai_prepare(struct snd_pcm_substream *substream)
 
 	dpcm_set_fe_update_state(fe, stream, SND_SOC_DPCM_UPDATE_FE);
 
-	/* there is no point preparing this FE if there are no BEs */
-	if (list_empty(&fe->dpcm[stream].be_clients)) {
-		dev_err(fe->dev, "ASoC: no backend DAIs enabled for %s\n",
-				fe->dai_link->name);
-		ret = -EINVAL;
-		goto out;
+	/* support also FE without BEs */
+	if (!list_empty(&fe->dpcm[stream].be_clients)) {
+		ret = dpcm_be_dai_prepare(fe, substream->stream);
```

Comment:
@juimonen this is quite contradicting your commit message and the comment above. You mention there is no BE but you need a be_dai_prepare()?

Reply:
oh I get it now. you comment above is incorrect. You are calling be_dai_prepare() for the BE's connected to the FE but you also prepare the FE if there are no BE's which happens further down

Reply:
@ranjani changed the comment

Reply:
@juimonen Thanks. The change seems logical to me for our topology. The KBL topology with a ref capture pipeline didnt need this because it was actually capturing the IV feedback from the speaker.

But what happens in the case that the playback pipeline isnt started?

Reply:
@ranj063 if trying to record without playback I get:
arecord: pcm_read:2032: read error: Input/output error
In this case it can be interpreted "logical" or not? We don't have anything to record, so you can't record. I didn't see any oops or errors in dmesg either. We could return 0 samples but that's bigger thing to implement...